src/main/java/net/openhft/chronicle/core/cleaner/impl/reflect/ReflectionBasedByteBufferCleanerService.java

```diff
@@ -49,10 +50,13 @@
             clean = lookup.findVirtual(cleanerClass, "clean", MethodType.methodType(void.class));
         } catch (NoSuchMethodException | ClassNotFoundException | IllegalAccessException e) {
             // Don't want to record this in tests so just send to slf4j
-            Logger.getLogger(ReflectionBasedByteBufferCleanerService.class.getName())
-                    .warning("Make sure you have set the command line option " +
-                            "\"--illegal-access=permit --add-exports java.base/jdk.internal.ref=ALL-UNNAMED\" " +
-                            "to enable " + ReflectionBasedByteBufferCleanerService.class.getSimpleName());
+            final Logger logger = Logger.getLogger(ReflectionBasedByteBufferCleanerService.class.getName());
+            if (logger.isLoggable(Level.WARNING)) {
```

Comment:
Do we really check that? Usually it is assumed that WARNING is logged, even if it isn't there are worse problems than wasting some CPU cycles on string concat.

Reply:
I agree. I think this is just the general pattern to avoid string concatenation if logging is disabled.

Reply:
Did you check the codebase to see if we actually do that? I suggest we do not use this pattern for WARNING and ERROR but only for DEBUG and INFO, that's what everybody seems to be doing. WDYT?

Reply:
Probably. Let's keep it as it is but stick to the way you described in the future.

Reply:
We should only check log level for debug IMO, otherwise we are just adding too much code

Reply:
Anyone who suppresses WARNING is a madman